### PR TITLE
Added ability to set read deadline

### PR DESCRIPTION
**Pull request by oxtoacart**
cc: @myleshorton

### CODE REVIEW

**Comment by xtaci** [comment]
@AudriusButkevicius

File: stream.go [hunk 1]
```diff
@@ -29,15 +37,26 @@ func newStream(id uint32, frameSize int, sess *Session) *Stream {
 	s.frameSize = frameSize
 	s.sess = sess
 	s.die = make(chan struct{})
+	s.readDeadline = maxTime
```
Comment (xtaci) [inline stream.go]: s.readDeadline should by default be zero.

File: stream.go [hunk 2]
```diff
@@ -29,15 +37,26 @@ func newStream(id uint32, frameSize int, sess *Session) *Stream {
...
 	return s
 }
 
 // Read implements io.ReadWriteCloser
 func (s *Stream) Read(b []byte) (n int, err error) {
+	var deadline <-chan time.Time
+	d := atomic.LoadInt64(&s.readDeadline)
+	if d > 0 {
+		timer := time.NewTimer(time.Duration(d - time.Now().UnixNano()))
+		defer timer.Stop()
```
Comment (AudriusButkevicius) [inline stream.go]: Just use time.After

File: stream.go (hunk 1 above)
Comment (oxtoacart) [inline stream.go]: Good catch, fixed!

**Comment by xtaci** [comment]
$ gometalinter .
stream.go:13:1:warning: maxTime is unused (deadcode)
stream.go:13:1:warning: zeroTime is unused (deadcode)

can you remove that?

**Comment by oxtoacart** [comment]
@xtaci Removed dead code.

BTW, thanks for this awesome library!

**Comment by xtaci** [comment]
cool thanks

**Pull request merged**